src/mongo/db/service_entry_point_common.cpp

```diff
@@ -1311,11 +1312,16 @@ void ExecCommandDatabase::_initiateCommand() {
     });
 
     rpc::readRequestMetadata(opCtx, request, command->requiresAuth());
-    uassert(ErrorCodes::Unauthorized,
+    if (auto token = _invocation->securityToken()) {
+        verifySecurityToken(*token, request.securityToken);
+        _tokenAuthorizationSessionGuard.emplace(opCtx, token);
+    } else if (auto dollarTenant = _invocation->ns().tenantId()) {
+        verifyDollarTenantField(opCtx);
+    }
```

Comment:
I'm grabbing the security token from the command invocation itself, and then manually setting it on the scope guard rather than putting it on the opCtx. Any other ideas here are most welcome!

Also, from an auth/authz POV I think we need a way to identify a tenant id as having come from $tenant as opposed to having come from a db prefix (if we do still support tenant ids coming in as prefixes until we swap out Atlas Proxy for mongoq).